### ISSUE TITLE
Blocks stop connecting after spamming the blocks in toolbox

### ISSUE THREAD

**Issue by msintov** [LLK/scratch-blocks#890]
### Expected Behavior

Blocks dragged to the workspace should be able to connect to an existing script.

### Actual Behavior

After spamming two blocks in the toolbox a number of times, the next time you drag a block to the workspace, it will not connect to an existing script.

### Steps to Reproduce

1. Load horizontal_playground.html in iOS WebView on a device.
2. Drag the green flag to the workspace. Connect an envelope block to it.
3. Tap two buttons in the toolbox alternating between them. For instance, very quickly tap message hat block once then envelope block once, alternating your taps between the blocks, a total of ~15 times.
4. Drag the envelope block to the workspace and try to connect to the green flag block.

Note: After this bug reproduces and you see the block doesn't connect, if you then pan the workspace, then the next time you try to drag a block to the workspace (such as the envelope hat block), the workspace pans at the same time that you are trying to drag the block.

### Operating System and Browser

iOS 10. Not yet tested on other OSes.

**Comment by griffpatch** [comment]
This bug is also manifest in another way, again it requires you to drag two or more connected blocks.
Grab two connected blocks and drag them over and up and down another stack of scripts. The blocks begin by appearing to accept the drag, but after doing this once the blocks under the drag stop accepting the drag. If you then drop the dragged blocks elsewhere you will find that the stack of blocks you were dragging over have become detached from each other (even though they are all still lined up like they are attached). You can test this by dragging the top of the stack you were dragging over and you will find they are all separated. This is therefore a particularly bad bug as it will destroy the existing scripts.

**Comment by paulkaplan** [comment]
Sorry I'm trying to reproduce this, but I can't seem to. @griffpatch are you saying you can reproduce on a desktop device or is that still only on mobile? Is it possible to make a screen recording or similar?

**Comment by msintov** [comment]
@paulkaplan Now that I've tested with the latest code, I don't reproduce my issue anymore. Hurrah! I suspect it was fixed by Rachel's "New dragging!" merge from Google on May 22, 2017.

**Comment by rachel-fenichel** [comment]
Huh, apparently rewriting all of dragging was good for something.  Let us know if this recurs.  (@griffpatch)

**Comment by griffpatch** [comment]
Here's a screen capture of me showing the behavior of dragging two blocks over a stack in a chrome browser. Notice how 'all' blocks are detached once the drag is complete even though I didn't release the drag over the stack? This is happening on https://llk.github.io/scratch-gui/ has this page got the latest changes from github?

![scratch_example](https://cloud.githubusercontent.com/assets/6737342/26724627/6cb209ba-4792-11e7-8535-7db61f9e0f1c.gif)

Griffpatch

**Comment by griffpatch** [comment]
Does this need re-opening or a new issue raising for it?

**Comment by msintov** [comment]
@griffpatch Since I believe my issue is fixed, I think it makes sense to open a new issue for yours.

**Comment by griffpatch** [comment]
New issue raised: #922
